Stratis.Bitcoin.Features.MemoryPool/MempoolManager.cs

```diff
@@ -15,7 +15,7 @@ namespace Stratis.Bitcoin.Features.MemoryPool
     /// <summary>
     /// A lock for managing asynchronous access to memory pool.
     /// </summary>
-    public class MempoolAsyncLock : AsyncLock
+    public class MempoolAsyncLock : SchedulerLock
```

Comment:
Maybe rename to `MempoolSchedulerLock` ?

Reply:
+1